pallets/gear/src/manager/task.rs

```diff
@@ -236,6 +214,38 @@ where
 
         // Consuming gas handler for waitlisted message.
         Pallet::<T>::consume_and_retrieve(waitlisted.id());
+
+        if waitlisted.kind() == DispatchKind::Init {
```

Comment:
isn't it better before consume and retrieve just call api from Journal handler impl?

Reply:
Frankly speaking we cannot since the impl isn't a standalone function of journal-handler. Nonetheless you've made a good remark so will extract the function to avoid copy-paste.

However I'm curious why do you suggest to process init-message before consume/retrive?